Task.1.2.Matrix/Task.1.2.Matrix/Entities/DiagonalMatrix.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task1.Two.Matrix.Entities
+{
+    /// <summary>
+    /// A generic class to create data structure in Diagonal matrix
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    internal class DiagonalMatrix<T>
```

Comment:
Inherit from SquareMatrix and reuse its methods, fileds and properties. 
The only difference between them if that Diagonal index setter does not do anything when i != j and getter returns default(T) when i != j.